catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +133,72 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
+              queryResponseTransformers) {
+
+            LOGGER.trace("Finding transformer to transform query response.");
+
+            String id = (String) queryResponseTransformer.getProperty("id");
+
+            if (id.equals(transformerId)) {
+              BinaryContent content =
+                  bundleContext
```

Comment:
❓ I'm not 100% on this, but I think we can actually just get a list of service references based on a filter that filters them to only be what we're looking for. `BundleContext` is a little broad in my opinion and if we could avoid it that would be good.

Reply:
Since the transformer ID is read off of the incoming request; I don't think filter parameter can be used in blueprint.  This seems to be the best approach to my knowledge to conditionally look up the correct service references.

Reply:
Well you could use `bundleContext.getServiceReferences(QueryResponseTransformer.class.getName(), "(id=<transformerId>)")` and not have to loop over the list of service references.